streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -437,7 +438,10 @@ private void rewriteSingleStoreSelfJoin(
         if (currentNode instanceof StreamStreamJoinNode && currentNode.parentNodes().size() == 1) {
             final StreamStreamJoinNode joinNode = (StreamStreamJoinNode) currentNode;
             // Remove JoinOtherWindowed node
-            final GraphNode parent = joinNode.parentNodes().stream().findFirst().get();
+            final GraphNode parent = joinNode.parentNodes().stream()
```

Comment:
It seems we checked there is one item in parentNodes in the `if` condition just above, so do we really need this?

Reply:
As you said, it doesn't seem necessary.